numbat/src/parser.rs

```diff
@@ -1724,6 +1730,12 @@ mod tests {
         should_fail(&["0x1.2", "0b1.0", "0o1.0", "0x.1", "0b.0", "0o.1"]);
     }
 
+    #[test]
```

Comment:
I had some trouble writing a proper test for `NaN` since `NAN == NAN` is `false`.

Reply:
You could add a test for this somewhere in the interpreter tests, and check that `NaN == NaN` evaluates to false in Numbat.

Reply:
There is a [test](https://github.com/sharkdp/numbat/blob/32c3e075f14d5682addd45208d5e88a1b8b617fd/numbat/tests/interpreter.rs#L112)  in the interpreter tests already which works fine but I thought it might be good to have a parser test too?

Reply:
Oh I missed that, but we can have something like `NaN != NaN` as a test too I suppose.

Reply:
I would think we're good with the `inf` test here. The code for NaN is the same.